### DISCUSSION TITLE
[@types/aws-lambda] version 8.10.119 to 8.10.120 breaks the existing system

### DISCUSSION THREAD

**Discussion by ErtanTaner** [DefinitelyTyped/DefinitelyTyped#68038] [Issues with a @types package]
## Description
In 8.10.19, sns.d.ts has a type named "SNSMessage" and it has a prop named "SigningCertURL". At the 8.10.120, it is changed to "SigningCertURL" which breaks the existing system which uses 8.10.119 but shouldn't because it's a patch version.

### 8.10.119
![Ekran Resmi 2023-12-29 16 56 45](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/62712555/c5335923-7dc1-4166-8710-eb99dbf8e32c)

### 8.10.120
![Ekran Resmi 2023-12-29 16 57 20](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/62712555/c2a5107a-f0f0-4230-a168-e72de66e3516)

[a comment by an automated account is not shown]

**Comment by darbio** [comment]
@ErtanTaner this package doesn't follow semver.

Does this change break your build?

**Reply by ErtanTaner** [reply]
I am sorry, ı didn't saw it. Yes, it broke my system. I was using 8.10.119 with carrot symbol so it tries to install the newer version. When type changed, tsc stopped working. I found a workaround but just want to let you know.

**Reply by darbio** [reply]
Sorry about that. It was implemented in a change a while ago to reflect the documentation. Are you successfully using the old (capitalised) URL? If so, I'd be keen to see your code base. 

When we approved, I couldn't find any reference to it being capitalised and the docs show it as being Url as opposed to URL.

**Comment by simonbuchan** [comment]
The rule is roughly "if it didn't work at runtime before, it's a bugfix, not a breaking change"

The major and minor versions were meant to follow the Lambda Node runtime version, which previously affected the Handler signature in a slightly breaking change way a couple of times. It's fairly meaningless now.

**Reply by jamesorlakin** [reply]
Indeed, `@types/aws-lambda` is slightly unique in that its tracking a live service that doesn't really have any official versioning. This package generally attempts to reflect the types seen in practice. AWS documentation can sometimes be a bit confusing. The change in question was introduced by https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66410 which has a good explanation as to why - in effect the capitalised value was wrong, the payload delivered to Lambda by SNS is subtly different than SNS' own native JSON.

This means anybody trying to use `SigningCertURL` in a SNS-triggered Lambda would've ran into `undefined` issues at runtime. That's a bug that needed patching as it's theoretically broken for everyone.

I'm sure there's users of this package using types outside of a Lambda function for convenience, but that's not the intended use and this is a great example of where it might not work. (It might be worth us putting a little JSDoc note on `SNSMessage` mentioning it's not the same as the native SNS HTTP API?)

Having said that, if you _have_ run into a scenario where your Lambda function has received `SigningCertURL` instead of `SigningCertUrl` we'd be very much interested! It would be an unfortunate inconsistency from AWS, but we could then have both in the type definitions with a JSDoc comment explaining when one might be set instead of the other.
